### PR TITLE
feature: handle maven-invoker-plugin reports as JUnit ones (JENKINS-70561

### PR DESCRIPTION
**Pull request by bguerin**
See https://issues.jenkins.io/browse/JENKINS-70561

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: jenkins-plugin/pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.346.x</artifactId>
+        <artifactId>bom-2.361.x</artifactId>
```
Comment (jglick) [inline jenkins-plugin/pom.xml:50]: Clearer to merge #587 first.

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/InvokerRunsPublisher.java [hunk 2]
```diff
@@ -31,6 +31,9 @@
 import hudson.model.Run;
 import hudson.model.StreamBuildListener;
 import hudson.model.TaskListener;
+import hudson.tasks.junit.JUnitResultArchiver;
+
+import org.apache.commons.lang3.BooleanUtils;
```
Comment (jglick) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/InvokerRunsPublisher.java:36]: Careful with this dependency if it is not being picked up via an explicit dep on the `commons-lang3-api` library plugin. Trivial uses like this are best avoided. https://github.com/jenkinsci/jenkins/pull/6267

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JUnitUtils.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    listener.getLogger().println(
+                            "[withMaven] " + publisherName + " - " + testResultSummary.getFailCount() + " unit test failure(s) found, mark job as unstable");
+                }
+                node.addAction(new WarningAction(Result.UNSTABLE).withMessage(testResultSummary.getFailCount() + " unit test failure(s) found"));
+                run.setResult(Result.UNSTABLE);
+            }
+        } catch (RuntimeException e) {
+            listener.error("[withMaven] " + publisherName + " - exception archiving JUnit results " + testResults + ": " + e + ". Failing the build.");
+            LOGGER.log(Level.WARNING, "Exception processing " + testResults, e);
+            run.setResult(Result.FAILURE);
```
Comment (jglick) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JUnitUtils.java:105]: This is an antipattern. If your Pipeline step encounters a random exception, just throw it up. Do not call `setResult(FAILURE)`.

(`setResult(UNSTABLE)` as in the `junit` plugin is also somewhat problematic but there is not a great replacement.)

**Comment by bguerin** [comment]
@jglick 

> Could use test coverage I think?

Of course !! I just loose my head every time I go into a Jenkins plugin ...

Most of the tests here are not Unit test, but Integration test, launching a release Jenkins, with its plugins, ...
While needed and maybe more relevant than unit ones, the are more complex to write because of all the environment to set up

How do you write "real" unit test in a plugin, how do you achieve dependency injection in production code so you can use mocks in unit tests ? I spend time looking at "major" plugins (Configuration as code, Docker, VSphere cloud, Warnings NG, Coverage API, ...) I cannot find the way to go

File: jenkins-plugin/pom.xml (hunk 1 above)
Comment (bguerin) [reply jenkins-plugin/pom.xml]: my initial plan was to merge https://github.com/jenkinsci/pipeline-maven-plugin/pull/587 first, but it needs more work. This PR does not change or rely on newer baseline

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JUnitUtils.java (hunk 3 above)
Comment (bguerin) [reply jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JUnitUtils.java:105]: you're right and I agree. For now the way the plugin is coded, it tries to be as discrete as possible to let the focus on the maven execution itself (to be clear, it catches exception everywhere). I know that you will answer that if someone use this plugin, it is its wish, one could just ` sh { mvn .... }`  and not use this plugin at all. So if we use this plugin and it has a problem, it should go to the surface to fill an issue and be fixed.

I will work on it to make all this right

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/InvokerRunsPublisher.java (hunk 2 above)
Comment (bguerin) [reply jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/InvokerRunsPublisher.java]: Rewrote using vanilla java

**Comment by bguerin** [comment]
> Could use test coverage I think?

Tests added

Many thanks @jglick for your review !

**Pull request merged**